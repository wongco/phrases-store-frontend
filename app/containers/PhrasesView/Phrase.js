import React from 'react';
import PropTypes from 'prop-types';

const Phrase = props => {
  const { text } = props;
  return <li>{text}</li>;
};

Phrase.propTypes = {
  text: PropTypes.string,
};

export default Phrase;
