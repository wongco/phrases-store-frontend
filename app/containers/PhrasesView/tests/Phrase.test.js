import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Phrase from '../Phrase';

describe('<Phrase />', () => {
  it('renders without crashing', () => {
    shallow(<Phrase />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<Phrase />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const text = 'inputText';
    const wrapper = mount(<Phrase text={text} />);
    const PhraseComponent = wrapper.find(Phrase);
    expect(PhraseComponent.prop('text')).toBe(text);
  });

  it('should render an li element', () => {
    const wrapper = mount(<Phrase />);
    const PhraseComponent = wrapper.find('li');
    expect(PhraseComponent).toHaveLength(1);
  });
});
