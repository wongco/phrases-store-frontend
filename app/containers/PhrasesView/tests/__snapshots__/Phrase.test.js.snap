// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Phrase /> matches snapshot for default criteria 1`] = `<li />`;
